File: TextosJapones/src/main/java/org/jisho/textosJapones/controller/LegendasController.java
package org.jisho.textosJapones.controller;

import java.net.URL;
import java.util.ResourceBundle;

import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.layout.AnchorPane;
import javafx.scene.layout.StackPane;

public class LegendasController implements Initializable {

	@FXML
	private AnchorPane apRoot;

	@FXML
	private StackPane stackPane;

	@FXML
	protected AnchorPane apConteinerRoot;

	@FXML
	private LegendasImportarController importarController;

	@FXML
	private LegendasGerarVocabularioController processarController;

	public AnchorPane getRoot() {
		return apConteinerRoot;
	}

	public StackPane getStackPane() {
		return stackPane;
	}

	public void initialize(URL arg0, ResourceBundle arg1) {
		importarController.setControllerPai(this);
		processarController.setControllerPai(this);
	}

	public static URL getFxmlLocate() {
		return LegendasController.class.getResource("/view/Legendas.fxml");
	}

	public static String getIconLocate() {
		return "/images/icoTextoJapones_128.png";
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/controller/legendas/LegendasController.kt
package br.com.fenix.processatexto.controller.legendas

import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.layout.AnchorPane
import javafx.scene.layout.StackPane
import java.net.URL
import java.util.ResourceBundle


class LegendasController : Initializable {

    @FXML
    private lateinit var apRoot: AnchorPane

    @FXML
    lateinit var stackPane: StackPane

    @FXML
    protected lateinit var apConteinerRoot: AnchorPane

    @FXML
    private lateinit var importarController: LegendasImportarController

    @FXML
    private lateinit var processarController: LegendasVocabularioController

    @FXML
    private lateinit var marcasController: LegendasMarcasController
    
    val root: AnchorPane get() = apConteinerRoot

    override fun initialize(arg0: URL?, arg1: ResourceBundle?) {
        importarController.controllerPai = this
        processarController.controllerPai = this
        marcasController.controllerPai = this
    }

    companion object {
        val fxmlLocate: URL get() = LegendasController::class.java.getResource("/view/legendas/Legendas.fxml") as URL
    }
}